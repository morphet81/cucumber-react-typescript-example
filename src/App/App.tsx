import './App.scss'
import { Button, Card, Grid, TextField, Typography } from '@mui/material'
import { ChangeEvent, useState } from 'react'
import api from '../service/api'

interface MovieInformation {
    Poster: string
    Title: string
    Type: string
    Year: string
    imdbID: string
}

interface OMDdSearchResult {
    Response: string
    Search: MovieInformation[]
    totalResults: number
}

interface MovieItemProps {
    movie: MovieInformation
}

const MovieItem = (props: MovieItemProps) => {
    const { movie } = props

    return (
        <Grid item>
            <Card>
                <Grid container direction="row">
                    <Grid width={80} item>
                        <img width="100%" src={movie.Poster} alt={`${movie.Title} poster`} />
                    </Grid>
                    <Grid container item direction="column">
                        <Typography>{movie.Title}</Typography>
                        <Typography>{movie.Year}</Typography>
                    </Grid>
                </Grid>
            </Card>
        </Grid>
    )
}

function App() {

    const [searchTerm, setSearchTerm] = useState<string>()
    const [movies, setMovies] = useState<MovieInformation[]>([])

    const handleSearchTermChange = (event: ChangeEvent<HTMLInputElement>) => {
        setSearchTerm(event.target.value)
    }

    const handleSearchClick = () => {
        api.get(``, {
            params: {
                s: searchTerm
            }
        })
            .then((response) => response.data)
            .then((result: OMDdSearchResult) => {
                return result.Search
            })
            .then(setMovies)
    }

    return (
        <Grid container direction="column" spacing={2}>
            <Grid item>
                <Typography variant='h1'>Movie App</Typography>
            </Grid>
            <Grid container item direction="row" alignItems="center" spacing={2}>
                <Grid item>
                    <TextField id="outlined-basic" label="Title, Actor, etc..." variant="outlined" onChange={handleSearchTermChange} />
                </Grid>
                <Grid item>
                    <Button variant="contained" onClick={handleSearchClick}>Search</Button>
                </Grid>
            </Grid>
            <Grid container item direction="column" spacing={2}>
                {movies.map(movie => {
                    return <MovieItem movie={movie} key={movie.imdbID} />
                })}
            </Grid>
        </Grid >
    )
}

export default App
