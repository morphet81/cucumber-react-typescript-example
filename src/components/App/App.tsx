import './App.module.scss'
import { Button, Grid, TextField, Typography } from '@mui/material'
import { ChangeEvent, KeyboardEvent, useState } from 'react'
import api from '../../service/api'
import MovieListItem from '../MovieListItem/MovieListItem'

interface OMDdSearchResult {
    Response: string
    Search: MovieInformation[]
    totalResults: number
}

function App() {

    const [searchTerm, setSearchTerm] = useState<string>()
    const [movies, setMovies] = useState<MovieInformation[]>([])

    const handleSearchTermChange = (event: ChangeEvent<HTMLInputElement>) => {
        setSearchTerm(event.target.value)
    }

    const handleSearchClick = () => {
        searchForMovies()
    }

    const handleSearchOnKeyDown = (event: KeyboardEvent) => {
        switch (event.key) {
            case `Enter`:
                searchForMovies()
                break
        }
    }

    const searchForMovies = () => {
        api.get(``, {
            params: {
                s: searchTerm,
            },
        })
            .then((response) => response.data)
            .then((result: OMDdSearchResult) => {
                return result.Search
            })
            .then(setMovies)
    }

    return (
        <Grid container direction="column" spacing={2}>
            <Grid item>
                <Typography variant='h1'>Movie App</Typography>
            </Grid>

            <Grid container item direction="row" alignItems="center" spacing={2}>
                <Grid item>
                    <TextField id="outlined-basic" label="Title, Actor, etc..." variant="outlined" onChange={handleSearchTermChange} onKeyDown={handleSearchOnKeyDown} />
                </Grid>
                <Grid item>
                    <Button variant="contained" onClick={handleSearchClick}>Search</Button>
                </Grid>
            </Grid>

            <Grid container item direction="column" spacing={2}>
                {movies.map(movie => {
                    return <MovieListItem movie={movie} key={movie.imdbID} />
                })}
            </Grid>
        </Grid >
    )
}

export default App
