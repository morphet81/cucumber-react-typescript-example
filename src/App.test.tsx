import { render, RenderResult } from '@testing-library/react'
import App from './App'

describe(`App`, () => {
    let container: RenderResult

    beforeEach(() => {
        container = render(<App />)
    })

    test(`matches previous screenshot`, () => {
        expect(container.baseElement).toMatchSnapshot();
    })

})