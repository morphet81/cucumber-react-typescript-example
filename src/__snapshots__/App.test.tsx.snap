// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches previous screenshot 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-1rl1og4-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
        >
          Movie App
        </h1>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 css-q9fxzi-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              Title, Actor, etc...
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-basic"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Title, Actor, etc...
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Search
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-10h8c1q-MuiGrid-root"
      />
    </div>
  </div>
</body>
`;
